ResearchKit/ActiveTasks/ORKGoNoGoViewController.m

```diff
@@ -0,0 +1,334 @@
+    for (NSNumber* go in tests) {
+        if ([go boolValue] == NO) {
+            hasNoGo = YES;
+        }
+    }
+    
+    // If not, put one in
+    if (!hasNoGo && tests.count > 1) {
+        [tests setObject:[NSNumber numberWithBool:NO] atIndexedSubscript:arc4random_uniform(tests.count - 1) + 1];
+    }
+    
+    self.activeStepView.activeCustomView = _gonogoContentView;
+    self.activeStepView.stepViewFillsAvailableSpace = YES;
+    [_gonogoContentView setStimulusHidden:YES];
+}
+
+- (void)viewDidAppear:(BOOL)animated {
+    [super viewDidAppear:animated];
+    [self start];
+    _shouldIndicateFailure = YES;
+}
+
+- (void)viewWillDisappear:(BOOL)animated {
+    [super viewWillDisappear:animated];
+    _shouldIndicateFailure = NO;
+}
+
+#pragma mark - ORKActiveStepViewController
+
+- (void)start {
+    [super start];
+    [self startStimulusTimer];
+    
+}
+
+#if TARGET_IPHONE_SIMULATOR
+- (void)motionBegan:(UIEventSubtype)motion withEvent:(UIEvent *)event {
+    if (event.type == UIEventSubtypeMotionShake) {
+        [self attemptDidFinish:nil];
+    }
+}
+
+#endif
+
+
+- (ORKStepResult *)result {
+    ORKStepResult *stepResult = [super result];
+    stepResult.results = _results;
+    return stepResult;
+}
+
+- (void)applicationWillResignActive:(NSNotification *)notification {
+    [super applicationWillResignActive:notification];
+    _validResult = NO;
+    [_stimulusTimer invalidate];
+    [_timeoutTimer invalidate];
+}
+
+- (void)applicationDidBecomeActive:(NSNotification *)notification {
+    [super applicationDidBecomeActive:notification];
+    [self resetAfterDelay:0];
+}
+
+#pragma mark - ORKRecorderDelegate
+
+- (void)recorder:(ORKRecorder *)recorder didCompleteWithResult:(ORKResult *)result {
+    [self attemptDidFinish:result];
+}
+
+#pragma mark - ORKDeviceMotionRecorderDelegate
+
+- (void)deviceMotionRecorderDidUpdateWithMotion:(CMDeviceMotion *)motion {
+    CMAcceleration v = motion.userAcceleration;
+    double vectorMagnitude = sqrt(((v.x * v.x) + (v.y * v.y) + (v.z * v.z)));
+    if (vectorMagnitude > [self gonogoTimeStep].thresholdAcceleration) {
+        [self stopRecorders];
+    }
+}
+
+#pragma mark - ORKGoNoGoStepViewController
+
+- (ORKGoNoGoStep *)gonogoTimeStep {
+    return (ORKGoNoGoStep *)self.step;
+}
+
+- (void)configureTitle {
+    int successCount = 0;
+    int errorCount = 0;
+    NSTimeInterval lastReactionTime = 0;
+    
+    for (ORKGoNoGoResult* res in _results) {
+        if (res.incorrect == NO) {
+            successCount++;
+        } else {
+            errorCount++;
+        }
+        if (res.go && !res.incorrect) {
+            lastReactionTime = res.timeToThreshold;
+        }
+    }
+    
+    NSString *format = ORKLocalizedString(@"GONOGO_TASK_ATTEMPTS_FORMAT", nil);
+    NSString *text = [NSString stringWithFormat:format, ORKLocalizedStringFromNumber(@(successCount + 1)), ORKLocalizedStringFromNumber(@([self gonogoTimeStep].numberOfAttempts))];
+    
+    if (errorCount > 0)
+    {
```

Comment:
ResearchKit style guide is to put the curly bracket on the same line as the if statement.